### PR TITLE
*: Set the error of schema expired to retryable error

### PR DESCRIPTION
**Pull request by zimulala**
*: set the error of schema expired to retryable error
domain: rename an error name

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @shenli @coocood @tiancaiamao

File: session.go [hunk 1]
```diff
@@ -234,13 +239,30 @@ func (s *session) finishTxn(rollback bool) error {
 			s.txn.SetOption(kv.BinlogData, bin)
 		}
 	}
-	err := s.txn.Commit()
-	if err != nil {
+
+	if err := s.checkSchemaValid(); err != nil {
+		if !s.sessionVars.RetryInfo.Retrying {
+			err = s.Retry()
```
Comment (coocood) [inline session.go]: `domain.ErrInfoSchemaExpired` is not `kv.IsRetryableError`, `Retry` will only retry once.

File: session.go [hunk 2]
```diff
@@ -119,6 +118,7 @@ type session struct {
 	values             map[fmt.Stringer]interface{}
 	store              kv.Storage
 	history            stmtHistory
+	currRetryCnt       int // current retry times
```
Comment (coocood) [inline session.go]: This seems unnecessary.

**Comment by coocood** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**